### PR TITLE
Resolver minor tweaks

### PR DESCRIPTION
**Pull request by enyst**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

- fix a leftover occurrence of litellm, to use our LLM class instead
- removed unused param in `send_pull_request`
- added `additional_message` contents when the resolver comments, even if it failed parsing
- misc comments added as I was reading the code

---
**Link of any specific issues this addresses**

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:ee167a8-nikolaik   --name openhands-app-ee167a8   docker.all-hands.dev/all-hands-ai/openhands:ee167a8
```

### CODE REVIEW

**Review by malhotra5 — APPROVED** [review]
LGTM, thanks!

We're probably due for a larger refactor since its getting increasingly hard to follow the code 😵‍💫

**Comment by enyst** [comment]
Funny you say that, I mean sure one can always be on the lookout for improvements, but this code is much, _much_ more readable and easier to follow than other parts of the codebase! ❤️
